app/models/range_progresses.rb

```diff
@@ -0,0 +1,68 @@
+class RangeProgresses
+  def initialize(id, start_date, end_date)
+    @start_date = start_date
+    @end_date = end_date
+    @progresses = DailyProgress.where(practice_id: id, done_at: start_date..end_date)
+    @done_dates = @progresses.where('progress > 0').pluck(:done_at).sort
+  end
+
+  def format_each_week
+    (@start_date..@end_date).each_with_object([]) do |date, formated_dates|
+      if formated_dates.blank? || formated_dates[-1].last.wday == 6
+        formated_dates << [date]
+      else
+        formated_dates[-1] << date
+      end
+    end
+  end
+
+  def total_done
+    @done_dates.count
+  end
+
+  def longest_streak
+    return @longest if defined? @longest
+
+    dates_by_continuity = \
+      @done_dates.each_with_object([]) do |date, dates_by_continuity|
+        if dates_by_continuity.blank? || dates_by_continuity.last.last + 1.day != date
+          dates_by_continuity << [date]
+        else 
+          dates_by_continuity.last << date
```

Comment:
__[reek]__
* [DuplicateMethodCall] calls dates_by_continuity.last 2 times
* [FeatureEnvy] refers to dates_by_continuity more than self